ocs_ci/ocs/ocp.py

```diff
@@ -241,6 +242,26 @@ def apply(self, yaml_file):
         command = f"apply -f {yaml_file}"
         return self.exec_oc_cmd(command)
 
+    def patch(self, resource_name, params):
+        """
+        Applies changes to resources
+
+        Args:
+            resource_name (str): Name of the resource
+            params (dict): Changes to be added to the resource
+
+        Returns:
+            bool: True in case if changes are applied. False otherwise
+
+        """
+        params = "\'" + json.dumps(params) + "\'"
```

Comment:
Using json.dumps leads to problem in case of different usage pattern. 
Eg: There are different type of operations, in this case it is --type 'json' but there is 'merge' which will fail.
so I would suggest to use 
params = "\'" + params + "\'"
& add one more param "type" in the function
def patch(self, resource_name, params, type='json'):
in line 258 provide the type.
command = f"patch {self.kind} {resource_name} -n {self.namespace} -p {params} --type {type}"

Reply:
Function:
```    def patch(self, resource_name, params, type='json'):
        """
        Applies changes to resources
        Args:
            resource_name (str): Name of the resource
            params (dict): Changes to be added to the resource
            type (str): Type of the operation
        Returns:
            bool: True in case if changes are applied. False otherwise
        """
        # params = "\'" + json.dumps(params) + "\'"
        params = "\'" + params + "\'"
        command = f"patch {self.kind} {resource_name} -n {self.namespace} -p {params} --type {type}"
        log.info(f"Command: {command}")
        result = self.exec_oc_cmd(command)
        if 'patched' in result:
            return True
        return False```
Call:
```assert ocp_obj.patch(
            resource_name=constants.IMAGE_REGISTRY_RESOURCE_NAME,
            params='{"spec": {"defaultRoute": true}}', type='merge'
        ), f"Registry pod defaultRoute enable is not success"```

Reply:
done